### PR TITLE
1.17.X port, (uses changes from #2814)

### PR DESCRIPTION
**Pull request by wagyourtail**
depends on #2814 

this is baritone 1.17, forge is dropping MCP mappings in favor of Mojmap so I remapped everything... I'm usually 99% accurate on those hence the couple extra commits. can't guarantee I got everything right for the 1.17 changes but every mapping I had to fix manually was checked against a mapping viewer with MCP and Mojmap.

# **__TODO__**
- [x] fix data-driven world heights (so any datapack that changes that, currently causes pathing/movement failures). shouldn't be too hard
- [x] verify chunk-cache still works/update for data-driven world height.
- [x] verify and fix rendering (line width might not be applying...)
- [x] make the dockerfile and stuff use java16
- [x] someone said behavior with new geode crystals was weird... check that, probably have to add to `isBlockNormalCube` or something

<@ me on discord or comment to add more, please use the unoptimized jar for stacktraces xd>

# **__Setting up the dev environment__**
This is on top of #2814, meaning it uses a fabric-style toolchain, you can (optionally because you should be using intellij) do `gradle genSources`

building is just
`./gradlew build` for vanilla/launchwrapper (to actually use, get optifine's patched [launchwrapper:2.0](https://github.com/sp614x/optifine/files/2342731/NewerJavaCompatibility.zip) to be compatible with java 16

`./gradlew build -Pbaritone.fabric_build` for fabric.
and
`./gradlew build -Pbaritone.forge_build -Ploom.forge`  once forge is released (with mixin support)...

for running in dev you may have to change the classpath to "launch" manually

 pr to my fork (and then @ me on discord) if you want to help contribute to the todo list

Build in artifacts zip at You can access it at https://github.com/cabaletta/baritone/actions/runs/1324383080

### CODE REVIEW

**Comment by AngelFireLA** [comment]
build fails with gradle 7.1.1 an djava 16 :/

File: src/main/java/baritone/pathing/movement/MovementHelper.java [hunk 1]
```diff
@@ -522,19 +548,19 @@ static boolean isBlockNormalCube(BlockState state) {
                 || block instanceof ShulkerBoxBlock) {
```
Comment (substanc3-dev) [inline src/main/java/baritone/pathing/movement/MovementHelper.java:548]: PointedDripstoneBlock needs to be added as an exception as well.

**Comment by substanc3-dev** [comment]
> 
> 
> build fails with gradle 7.1.1 and java 16 :/

builds fine for me

File: src/main/java/baritone/pathing/movement/MovementHelper.java (hunk 1 above)
Comment (wagyourtail) [reply src/main/java/baritone/pathing/movement/MovementHelper.java:548]: maybe the avoid list would be better...

Comment (substanc3-dev) [reply src/main/java/baritone/pathing/movement/MovementHelper.java:548]: not sure what you mean, but the getCollisionShape call NullPointerExceptions on PointedDripstoneBlock

Comment (wagyourtail) [reply src/main/java/baritone/pathing/movement/MovementHelper.java:548]: ahh. yeah ok. I was thinking add it to movement helper line 120 as well.

**Comment by monkeyboyt** [comment]
I'm confused this all means that you can't make 1.17 right?

**Comment by wagyourtail** [comment]
> I'm confused this all means that you can't make 1.17 right?

this is work in progress 1.17... it builds for fabric and vanilla but has some things that are currently broken, hence "work in progress"

**Comment by leijurv** [comment]
What subversion of 1.17 is this for? I'll make you a branch.

**Comment by wagyourtail** [comment]
> What subversion of 1.17 is this for? I'll make you a branch.

1.17.1

**Comment by leijurv** [comment]
Created https://github.com/cabaletta/baritone/tree/1.17.1

**Comment by Nova-69** [comment]
#mine works perfectly on single player, but not on a server like Hypixel, even with LegitMine on. Looking forward to this projects progress, good luck  👌

**Comment by scorbett123** [comment]
> #mine works perfectly on single player, but not on a server like Hypixel, even with LegitMine on. Looking forward to this projects progress, good luck  👌 

Baritone is not meant to be used on servers that it is not allowed.

**Comment by wagyourtail** [comment]
turns out I can't re-name my branch and keep the same pr...

**Comment by wagyourtail** [comment]
forge doesn't have mixin for 1.17 (and shedaniels hasn't finished arch-loom's 1.17 forge support yet) so, I don't think I can add forge yet...

**Comment by loadingfd** [comment]
it doesn't work with the datapack of cave, will it be added in the future?

**Comment by wagyourtail** [comment]
Yeah, that's already on the todo list

**Comment by JanzenJohn** [comment]
Doesn't build for me, ":proguard" complains about not having java 16 but don't you need java 16 to even get to ProGuard ?

is there a way I can set the java version ProGuard uses ? [My JAVA_HOME points to a java16 install btw]

**Comment by AuracleTech** [comment]
#build gets stuck due to the orientation of the blocks in the schematics
can't find a way around it other than not use any block that aren't purely solid non orientable blocks

https://user-images.githubusercontent.com/40531871/127770243-308ad1a8-bcde-489f-882b-167ffd0d6644.mp4

**Comment by codecnotsupported** [comment]
>     * fix data-driven world heights (so any datapack that changes that, currently causes pathing/movement failures). shouldn't be too hard

Having no prior experience with java I attempted to resolve this by offsetting the height varibles the constructors of betterblockpos by the appropriate offset.
goto x,z works-ish?
Any idea what a proper solution would be?

**Comment by wagyourtail** [comment]
* find all the places where baritone scan's the world and adjust it to work with data driven world heights
* adjust the cached chunk provider to accept/respond with the right world height
* make sure the pathing process then understands the new world heights and puts them at the correct height

**Comment by scorbett123** [comment]
You would probably need to slightly edit some of the goals due to them assuming that y will always be possible.

**Comment by scorbett123** [comment]
Sorry wrong button

**Comment by SupAnonymo** [comment]
## Some information
Operating system: Windows 10
Java version: Java 8 
Minecraft version: 1.17
Baritone version: baritone-standalone-fabric-1.7.0 / baritone-unoptimized-fabric-1.7.0
Other mods (if used): none

## Exception, error or logs
[latest.log](https://github.com/cabaletta/baritone/files/6951576/latest.log)


Linux: `~/.minecraft/logs/`

Windows: `%appdata%/.minecraft/logs/`

Mac: `/Library/Application\ Support/minecraft/logs/`

## How to reproduce
When I try to mine (#mine) baritone it doesn't walk around and render minerals on the surface, or in places where those minerals aren't. The server does not have Anti-xray.
Some of the screenshots were captured while using Meteor's x-ray, but I also uninstalled meteor and tried it and it didn't work either. The log I attached is without any mod except baritone.

## Modified settings
none

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have included the version of Minecraft I'm running, baritone's version and forge mods (if used).
- [x] I have included logs, exceptions and / or steps to reproduce the issue.
- [x] I have not used any OwO's or UwU's in this issue.
image
image
image
image
image
image

**Comment by wagyourtail** [comment]
this world uses the modified worldheight vanilla feature?

**Comment by loadingfd** [comment]
Whether you stop updating it ?

**Comment by Gh0st-Zer0** [comment]
## Some information
Operating system: Windows 10
Java version: 1.8.0_301-b09
Minecraft version: forge 1.17.1
Baritone version: baritone-standalone-forge-1.7.1
Other mods (if used): 

## Exception, error or logs

[crash-2021-10-08_20.15.27-client.txt](https://github.com/cabaletta/baritone/files/7314900/crash-2021-10-08_20.15.27-client.txt)

## What happened:
The game crashes as soon as I try to load a world or server with forge and baritone.

## How to reproduce
Try to join any world, online or offline, with baritone for forge I guess?

## Modified settings
To get the modified settings run `#modified` in game

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have included the version of Minecraft I'm running, baritone's version and forge mods (if used).
- [x] I have included logs, exceptions and / or steps to reproduce the issue.
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by pikachu1298** [comment]
you did it let's go!

**Pull request merged**